### PR TITLE
Implement `create_logged_model`

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/13239?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/13239/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 13239
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

title

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

```python
from databricks.sdk import WorkspaceClient

import mlflow

wc = WorkspaceClient()
me = wc.current_user.me()

mlflow.set_tracking_uri("databricks")
mlflow.set_experiment(f"/Users/{me.user_name}/mlflow-3-test")


with mlflow.start_run() as run:
    pass

client = mlflow.MlflowClient()
model = client.create_logged_model(
    experiment_id=run.info.experiment_id,
    run_id=run.info.run_id,
    name="test",
    params={"param": "param"},
    tags={"tag": "tag"},
    model_type="test",
)
print(model.to_proto())
```

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/entities/model_status.py [hunk 1]
```diff
@@ -1,9 +1,38 @@
...
+
+    def to_proto(self):
+        if self == ModelStatus.UNSPECIFIED:
+            return LoggedModelStatus.LOGGED_MODEL_STATUS_UNSPECIFIED
+        elif self == ModelStatus.PENDING:
+            return LoggedModelStatus.LOGGED_MODEL_PENDING
+        elif self == ModelStatus.READY:
+            return LoggedModelStatus.LOGGED_MODEL_READY
+        elif self == ModelStatus.FAILED:
+            return LoggedModelStatus.LOGGED_MODEL_FAILED
```
Comment (liangz1) [inline mlflow/entities/model_status.py:23]:
```suggestion
            return LoggedModelStatus.LOGGED_MODEL_UPLOAD_FAILED
```

File: mlflow/store/tracking/rest_store.py [hunk 2]
```diff
@@ -541,6 +553,42 @@ def record_logged_model(self, run_id, mlflow_model):
         )
         self._call_endpoint(LogModel, req_body)
 
+    def create_logged_model(
+        self,
+        experiment_id: str,
+        name: str,
+        run_id: Optional[str] = None,
```
Comment (liangz1) [inline mlflow/store/tracking/rest_store.py:560]:
```suggestion
        source_run_id: Optional[str] = None,
```
Shall we call it `source_run_id`? I'm thinking that adding the `source_` prefix prevents the user to assume that the logged_model object is **owned** by the run.

**Review by liangz1 — COMMENTED** [review]
Looks good! Just a heads up that the tags and params backend logic in CreateLoggedModel are still not checked in. params will be checked in later this week.
Also left some nit comments.

File: mlflow/store/tracking/rest_store.py [hunk 3]
```diff
@@ -541,6 +553,42 @@ def record_logged_model(self, run_id, mlflow_model):
...
+
+        Returns:
+            The created model.
+        """
+        req_body = message_to_json(
+            CreateLoggedModel(
+                experiment_id=experiment_id,
+                name=name,
+                model_type=model_type,
+                source_run_id=run_id,
```
Comment (harupy) [inline mlflow/store/tracking/rest_store.py:584]:
```suggestion
                source_run_id=source_run_id,
```

File: mlflow/store/tracking/rest_store.py (hunk 2 above)
Comment (harupy) [reply mlflow/store/tracking/rest_store.py:560]: Sounds good!

File: mlflow/entities/model_status.py (hunk 1 above)
Comment (harupy) [reply mlflow/entities/model_status.py:23]: Did we update protos?

Comment (harupy) [reply mlflow/entities/model_status.py:23]: never mind.

File: mlflow/store/tracking/rest_store.py (hunk 2 above)
Comment (harupy) [reply mlflow/store/tracking/rest_store.py:560]: @dbczumar what do you think?

Comment (dbczumar) [reply mlflow/store/tracking/rest_store.py:560]: Yep! `source_run_id` is a good name here, same for the entity

File: mlflow/entities/logged_model.py [hunk 4]
```diff
@@ -160,3 +161,40 @@ def to_dictionary(self) -> Dict[str, Any]:
...
+    def from_proto(cls, proto):
+        return cls(
+            experiment_id=proto.info.experiment_id,
+            model_id=proto.info.model_id,
+            name=proto.info.name,
+            artifact_location=proto.info.artifact_uri,
+            creation_timestamp=proto.info.creation_timestamp_ms,
+            last_updated_timestamp=proto.info.last_updated_timestamp_ms,
+            model_type=proto.info.model_type,
+            run_id=proto.info.source_run_id,
```
Comment (dbczumar) [inline mlflow/entities/logged_model.py:195]: Can we rename the field at the entity level to `source_run_id`? cc @liangz1

File: mlflow/store/tracking/abstract_store.py [hunk 5]
```diff
@@ -694,8 +695,8 @@ def create_logged_model(
         experiment_id: str,
         name: str,
         run_id: Optional[str] = None,
-        tags: Optional[Dict[str, str]] = None,
-        params: Optional[Dict[str, str]] = None,
+        tags: Optional[ModelTag] = None,
+        params: Optional[List[ModelParam]] = None,
```
Comment (dbczumar) [inline mlflow/store/tracking/abstract_store.py:699]: @harupy Do we have to update FileStore or existing MlflowClient method implementations for LogModels to account for this?

**Review by dbczumar — APPROVED** [review]
LGTM once comments are addressed!

File: mlflow/store/tracking/abstract_store.py (hunk 5 above)
Comment (harupy) [reply mlflow/store/tracking/abstract_store.py:699]: let me check

Comment (harupy) [reply mlflow/store/tracking/abstract_store.py]: This change comes from `FileStore`.

**Pull request merged**